src/main/java/de/teamlapen/vampirism/entity/minion/MinionEntity.java

```diff
@@ -610,4 +609,27 @@ private void handleLoadedMinionData(@NotNull T data) {
             this.discard();
         }
     }
+
+    /**
+     * serializes all allowed {@link net.minecraftforge.common.capabilities.Capability}s
+     */
+    protected CompoundTag serializeMinionCaps() {
+        Collection<String> allowedCapTags = getAllowedCapTags();
+        CompoundTag tag = this.serializeCaps();
+        if (tag != null) {
+            tag.getAllKeys().removeIf(s -> {
+                return !allowedCapTags.contains(s);
+            });
+            return tag;
+        } else {
+            return new CompoundTag();
+        }
+    }
+
+    /**
+     * @return all allowed capability identifiers
+     */
+    protected Collection<String> getAllowedCapTags() {
+        return Collections.singleton(new ResourceLocation("armourers_workshop", "entity-skin-provider").toString());
```

Comment:
if you wish to use a whitelist, I recommend using the data pack design.

Reply:
example:

in some mod, define a json like the tags:
resources/data/vampirism/minion.json 
```json
{
"capabillities": [
  "armourers_workshop:entity-skin-provider" 
 ]
}
```

Reply:
generally yes, but I currently do not see the need for it